### ISSUE TITLE
Incorrect error handling in  stub_blkgetsize()

### ISSUE THREAD

**Issue by mato** [mirage/mirage-block-unix#25]
The following code from stub_blkgetsize() has incorrect error handling:

````
  fd = open(filename_c, O_RDONLY, 0);
  if (blkgetsize(fd, &size_in_bytes) == 0)
    success = 0;
  close(fd);
  leave_blocking_section();

  free((void*)filename_c);

  if (fd == -1) uerror("open", filename);
  if (success == -1) uerror("BLKGETSIZE", filename);
````
If the call to open() fails then blkgetsize() will still be called (and fail with EBADF). Also, the two if() blocks in sequence will result in an error being reported from BLKGETSIZE rather than open.

**Comment by mato** [comment]
Note that this should be irrelevant if #24 is fixed, since the blkgetsize stub will not need to call open() at all.

**Comment by djs55** [comment]
Obsoleted by the fix for #24
